client/src/components/MapLayout/index.jsx

```diff
@@ -1,35 +1,75 @@
 import React, { Component } from 'react';
-import { Row, Col } from 'react-materialize'
+import {Row, Col, Input, Button} from 'react-materialize'
 import MapFilters from './MapFilters';
 import EventMap from '../EventMap';
 import { request } from '../../utils';
+import SelectCategory from '../EventForm/SelectCategory';
 
 
 class MapLayout extends Component {
   state = {
     events: [],
+    day_filter: 14,
+    category: ''
+  }
+
+  onChangeHandler = (e) => {
+      let {id} = e.currentTarget
+      this.setState({[id]: e.currentTarget.value})
+  };
+  filter = (e) => {
+      let filterData = {
+          day_filter: this.state.day_filter,
+          category: this.state.category
+      }
+      request('/events-short-info', 'POST', JSON.stringify(filterData)).then(data => {
+          if ('info' in data) {
+              this.setState({events: []})
+              data.info.forEach(element => {
+                  let eventShortInfo = this.state.events;
+                  eventShortInfo.push(element);
+                  this.setState({ events: eventShortInfo });
+                  console.log()
+              });
+          }
+      });
   }
 
   componentDidMount() {
-    request('/events-short-info').then(data => {
-      if ('info' in data) {
-        data.info.forEach(element => {
-          let eventShortInfo = this.state.events;
-          eventShortInfo.push(element);
-          this.setState({ events: eventShortInfo });
-        });
+      let filterData = {
+          day_filter: this.state.day_filter
+
       }
-    });
+      request('/events-short-info', 'POST', JSON.stringify(filterData)).then(data => {
+          if ('info' in data) {
+              data.info.forEach(element => {
+                  let eventShortInfo = this.state.events;
+                  eventShortInfo.push(element);
+                  this.setState({ events: eventShortInfo });
+
+              });
+          }
+      });
   }
 
   render(){
     return (
       <Row>
+          <Input id='day_filter' s={3} type='select' label="Filter by days" defaultValue='' onChange={this.onChangeHandler}>
```

Comment:
You have 14 days by default in the state, so I think it's better to set it by default here.